packages/calling/src/CallSettings/types.ts

```diff
@@ -90,15 +90,149 @@ export type CallSettingResponse = {
+ * This response object will have the status code, data and message and the data will have the callSetting object
+ * which can be of different types based on the API called.
+ *
+ * Example
+ * ```json
+ * {
+ *  statusCode: 200,
+ *    data: {
+ *      callSetting: ToggleSetting | CallForwardSetting | VoicemailSetting | CallForwardAlwaysSetting
+ *    },
+ *  message: 'SUCCESS'| 'FAILURE' | null
+ * }
+ * ```
+ *
+ */
 export interface ICallSettings {
-  getCallWaitingSetting: () => Promise<CallSettingResponse>;
-  getDoNotDisturbSetting: () => Promise<CallSettingResponse>;
-  setDoNotDisturbSetting: (flag: boolean) => Promise<CallSettingResponse>;
-  getCallForwardSetting: () => Promise<CallSettingResponse>;
-  setCallForwardSetting: (request: CallForwardSetting) => Promise<CallSettingResponse>;
-  getVoicemailSetting: () => Promise<CallSettingResponse>;
-  setVoicemailSetting: (request: VoicemailSetting) => Promise<CallSettingResponse>;
-  getCallForwardAlwaysSetting: (directoryNumber?: string) => Promise<CallSettingResponse>;
+  /**
+   * This API is used to fetch the call waiting setting.
+   *
+   * Example
+   * ```javascript
+   * const callWaitingResponse = await callSettings.getCallWaitingSetting();
+   * ```
+   *
+   * The callWaitingResponse object will have callSetting object with the properties as mentioned in ToggleSetting.
+   *
+   * Example - ToggleSetting
+   * ```json
+   * {
+   *  statusCode: 200,
+   *    data: {
+   *      callSetting: {
+   *        enabled: true,
+   *        ringSplashEnabled: true
+   *      },
+   *    },
+   *  message: null
+   * }
+   * ```
+   *
+   */
+  getCallWaitingSetting(): Promise<CallSettingResponse>;
+
+  /**
+   * This API is used to fetch the do not disturb(DND) status.
+   *
+   * Example
+   * ```javascript
+   * const dndResponse = await callSettings.getDoNotDisturbSetting();
+   * ```
+   *
+   * The dndResponse object will have callSetting object with the properties as mentioned in ToggleSetting.
+   *
+   * Example - ToggleSetting
+   * ```json
+   * {
+   *  statusCode: 200,
+   *    data: {
+   *      callSetting: {
+   *        enabled: true,
+   *        ringSplashEnabled: true
+   *      },
+   *    },
+   *  message: null
+   * }
+   * ```
+   */
+  getDoNotDisturbSetting(): Promise<CallSettingResponse>;
+
+  /**
+   * This API is used to set DND to true or false based on parameter received.
+   *
+   * Example
+   * ```javascript
+   * const dndResponse = await callSettings.setDoNotDisturbSetting(true|false);
+   * ```
+   */
+  setDoNotDisturbSetting(flag: boolean): Promise<CallSettingResponse>;
+
+  /**
+   * This API is used to fetch the call forward setting.
+   *
+   *  Example
+   * ```javascript
+   * const callForwardResponse = await callSettings.getCallForwardSetting();
+   * ```
+   *
+   * The callForwardResponse object will have callSetting object with the properties as mentioned in CallForwardSetting.
```

Comment:
The `CallForwardSettings` name should be matched here too.
```suggestion
   * The callForwardResponse object will have an object with the properties as mentioned in `CallSettingResponse`.
```